src/ResetMap.jl

```diff
@@ -0,0 +1,290 @@
+all other dimensions.
+"""
+function get_b(rm::ResetMap{N}) where {N<:Real}
+    n = dim(rm)
+    b = sparsevec(Int[], N[], n)
+    for (i, val) in rm.resets
+        b[i] = val
+    end
+    return b
+end
+
+
+# --- LazySet interface functions ---
+
+
+"""
+    dim(rm::ResetMap)
+
+Return the dimension of a reset map.
+
+### Input
+
+- `rm` -- reset map
+
+### Output
+
+The dimension of a reset map.
+"""
+function dim(rm::ResetMap)::Int
+    return dim(rm.X)
+end
+
+"""
+    σ(d::AbstractVector{N}, rm::ResetMap{N}) where {N<:Real}
+
+Return the support vector of a reset map.
+
+### Input
+
+- `d`  -- direction
+- `rm` -- reset map
+
+### Output
+
+The support vector in the given direction.
+If the direction has norm zero, the result depends on the wrapped set.
+"""
+function σ(d::AbstractVector{N}, rm::ResetMap{N}) where {N<:Real}
+    d_reset = copy(d)
+    for var in keys(rm.resets)
+        d_reset[var] = zero(N)
+    end
+    return substitute(rm.resets, σ(d, rm.X))
+end
+
+"""
+    ρ(d::AbstractVector{N}, rm::ResetMap{N}) where {N<:Real}
+
+Return the support function of a reset map.
+
+### Input
+
+- `d`  -- direction
+- `rm` -- reset map
+
+### Output
+
+The support function in the given direction.
+
+### Notes
+
+We use the usual dot-product definition, but for unbounded sets we redefine the
+product between ``0`` and ``±∞`` as ``0``; Julia returns `NaN` here.
+
+```jldoctest
+julia> Inf * 0.0
+NaN
+```
+
+See the discussion
+[here](https://math.stackexchange.com/questions/28940/why-is-infty-cdot-0-not-clearly-equal-to-0).
+"""
+function ρ(d::AbstractVector{N}, rm::ResetMap{N}) where {N<:Real}
+    return dot_zero(d, σ(d, rm))
+end
+
+"""
+    an_element(rm::ResetMap)
+
+Return some element of a reset map.
+
+### Input
+
+- `rm` -- reset map
+
+### Output
+
+An element in the reset map.
+It relies on the `an_element` function of the wrapped set.
+"""
+function an_element(rm::ResetMap)
+    return substitute(rm.resets, an_element(rm.X))
+end
+
+"""
+    isempty(rm::ResetMap)::Bool
+
+Return if a reset map is empty or not.
+
+### Input
+
+- `rm` -- reset map
+
+### Output
+
+`true` iff the wrapped set is empty.
+"""
+function isempty(rm::ResetMap)::Bool
+    return isempty(rm.X)
+end
+
+"""
+    constraints_list(rm::ResetMap{N}) where {N<:Real}
+
+Return the list of constraints of a (polyhedral) reset map.
+
+### Input
+
+- `rm` -- reset map
+
+### Output
+
+The list of constraints of the reset map.
+
+### Notes
+
+We assume that the underlying set `X` is polyhedral, i.e., offers a method
+`constraints_list(X)`.
+
+### Algorithm
+
+We fall back to `constraints_list` of a `LinearMap`, which results in
+zero-constraints for the dimensions that are reset.
+Then, for these dimensions, we modify the constraints by pushing them to their
+new reset value.
+"""
+function constraints_list(rm::ResetMap{N}) where {N<:Real}
+    function find_unique_nonzero_entry(vector::AbstractVector{N})
```

Comment:
Say what this function does:

```julia
# if `vector` has exactly one non-zero entry, return its index; in all other cases return 0 
function find_unique_nonzero_entry(vector::AbstractVector{N})
...
end
```
That said, note that it can be written as:

```julia
    return count(!iszero, vector) != 1 ? findfirst(!iszero, vector) : 0
```
But this function is worse than yours because it needs two passes on `vector`  😄